blockchain/setup.go

```diff
@@ -1,210 +1,121 @@
 	if err != nil {
-		return nil, fmt.Errorf("Error reading peer config: %v", err)
-	}
-	for _, p := range peerConfig {
-		if p.EventHost != "" && p.EventPort != 0 {
-			fmt.Printf("EventHub connect to peer (%s:%d)\n", p.EventHost, p.EventPort)
-			eventHub.SetPeerAddr(fmt.Sprintf("%s:%d", p.EventHost, p.EventPort),
-				p.TLS.Certificate, p.TLS.ServerHostOverride)
-			foundEventHub = true
-			break
-		}
+		return fmt.Errorf("failed to create chaincode package: %v", err)
 	}
 
-	if !foundEventHub {
-		return nil, fmt.Errorf("No EventHub configuration found")
+	// Install our chaincode on org peers
+	installCCReq := resmgmt.InstallCCRequest{Name: setup.ChainCodeID, Path: setup.ChaincodePath, Version: "1.0", Package: ccPkg}
+	_, err = setup.admin.InstallCC(installCCReq)
+	if err != nil {
+		return fmt.Errorf("failed to install cc to org peers %v", err)
 	}
 
-	return eventHub, nil
-}
+	// Set up chaincode policy
+	ccPolicy := cauthdsl.SignedByAnyMember([]string{"org1.hf.chainhero.io"})
 
-// Install and instantiate the chaincode
-func (setup *FabricSetup) InstallAndInstantiateCC() error {
-
-	// Check if chaincode ID is provided
-	// otherwise, generate a random one
-	if setup.ChaincodeId == "" {
-		setup.ChaincodeId = fcutil.GenerateRandomID()
-	}
-
-	fmt.Printf(
-		"Chaincode %s (version %s) will be installed (Go Path: %s / Chaincode Path: %s)\n",
-		setup.ChaincodeId,
-		setup.ChaincodeVersion,
-		setup.ChaincodeGoPath,
-		setup.ChaincodePath,
-	)
-
-	// Install ChainCode
-	// Package the go code and make a proposal to the network with this new chaincode
-	err := fcutil.SendInstallCC(
-		setup.Client, // The SDK client
-		setup.Channel, // The channel concerned
-		setup.ChaincodeId,
-		setup.ChaincodePath,
-		setup.ChaincodeVersion,
-		nil,
-		setup.Channel.GetPeers(), // Peers concerned by this change in the channel
-		setup.ChaincodeGoPath,
-	)
+	// Org resource manager will instantiate our chaincode on the channel
+	err = setup.admin.InstantiateCC(setup.ChannelID, resmgmt.InstantiateCCRequest{Name: setup.ChainCodeID, Path: setup.ChaincodePath, Version: "1.0", Args: [][]byte{[]byte("init")}, Policy: ccPolicy})
 	if err != nil {
-		return fmt.Errorf("Send install proposal return error: %v", err)
-	} else {
-		fmt.Printf("Chaincode %s installed (version %s)\n", setup.ChaincodeId, setup.ChaincodeVersion)
+		return fmt.Errorf("failed to instantiate the chaincode: %v", err)
 	}
 
-	// Instantiate ChainCode
-	// Call the Init function of the chaincode in order to initialize in every peer the new chaincode
-	err = fcutil.SendInstantiateCC(
-		setup.Channel,
-		setup.ChaincodeId,
-		setup.ChannelId,
-		[]string{"init"}, // Arguments for the invoke request
-		setup.ChaincodePath,
-		setup.ChaincodeVersion,
-		[]api.Peer{setup.Channel.GetPrimaryPeer()}, // Which peer to contact
-		setup.EventHub,
-	)
+	// Channel client is used to query and execute transactions
+	setup.client, err = setup.sdk.NewClient(fabsdk.WithUser("User1")).Channel(setup.ChannelID)
```

Comment:
Move the "User1" in configuration of the Setup struct

Reply:
@MaximeAubanel

Reply:
Done